app/Services/PolygonService.php

```diff
@@ -212,24 +215,25 @@ protected function validateSitePolygonProperties(string $polygonUuid, array $pro
      *
      * @return string UUID of resulting PolygonGeometry
      */
-    protected function transformAndStorePoints($geojson, $sitePolygonProperties): string
+    protected function transformAndStorePoints($geojson, $sitePolygonProperties): array
     {
-        $pointUuids = [];
-        foreach ($geojson['features'] as $feature) {
-            $pointUuids[] = $this->insertSinglePoint($feature);
+        foreach ($geojson['features'] as &$feature) {
+            $currentPointUUID = $this->insertSinglePoint($feature);
+            $feature['properties']['point_id'] = $currentPointUUID;
         }
 
         $properties = $sitePolygonProperties;
+        $mainSiteID = '';
         foreach (self::POINT_PROPERTIES as $property) {
             $properties[$property] = collect(data_get($geojson, "features.*.properties.$property"))->filter()->first();
+            if ($property === 'site_id') {
+                $mainSiteID = $properties[$property];
+            }
         }
 
-        // TODO:
-        //  * transform points into a polygon
-        //  * Insert the polygon into PolygonGeometry
-        //  * Create the SitePolygon using the data in $properties (including $properties['site_id'] to identify the site)
-        //  * Return the PolygonGeometry's real UUID instead of this fake return
+        $polygonsGeojson = App::make(PythonService::class)->voronoiTransformation($geojson);
+        $polygonsUuids = $this->createGeojsonModels($polygonsGeojson, ['site_id' => $mainSiteID]);
```

Comment:
You should be able to get this from `$properties['site_id']` instead of assigning it to a custom variable in the loop.

Also, I believe the `properties` should be assigned to all of the site polygons, not just the ones they happen to be set on; which is why that loop is there to begin with. I think this line should simply be (note I'm returning here, as the `polygonsUuids` variable is superfluous):
```
return $this->createGeojsonModels($polygonsGeojson, $properties);
```

Reply:
Here is now only using properties that come from point, due to the change in understanding that everypoint will have its own attributes for site polygon.